addons/sale_stock/models/account_invoice.py

```diff
@@ -52,7 +52,7 @@ def _compute_average_price(self, qty_done, quantity, moves):
                 qty_to_consider = invoiced_qty - qty_done
             qty_to_consider = min(qty_to_consider, quantity - qty_delivered)
             qty_delivered += qty_to_consider
-            average_price_unit = (average_price_unit * (qty_delivered - qty_to_consider) + move.price_unit * qty_to_consider) / qty_delivered
+            average_price_unit = (average_price_unit * (qty_delivered - qty_to_consider) - (move.value / move.product_qty) * qty_to_consider) / qty_delivered
```

Comment:
to clarify:
- move.price_unit
- move._get_price_unit()
- move.value

Actually, move._get_price_unit is intended to set the value on IN move

action_done on out move will write move.price_unit according to the cost method

so this change is probably useless